pkg/proxy/healthcheck/service_health.go

```diff
@@ -199,9 +196,9 @@ func (hcI *hcInstance) listenAndServeAll(hcs *server) error {
 
 func (hcI *hcInstance) closeAll() error {
 	errors := []error{}
-	for _, listener := range hcI.listeners {
-		if err := listener.Close(); err != nil {
-			klog.ErrorS(err, "Error closing listener for health check service", "service", hcI.nsn, "address", listener.Addr())
+	for _, server := range hcI.httpServers {
+		if err := server.Close(); err != nil {
```

Comment:
don't you want to check for `err != http.ErrServerClosed` here too?

Reply:
No, `Close` returns error returned from closing the Server's underlying listeners. It should be nil for normal case, and can never be `ErrServerClosed`.